src/Servers/Kestrel/test/FunctionalTests/Http2/ShutdownTests.cs

```diff
@@ -145,16 +145,14 @@ public async Task GracefulTurnsAbortiveIfRequestsDoNotFinish()
                 var stopServerTask = server.StopAsync(cts.Token).DefaultTimeout();
 
                 await closingMessageTask;
+
+                var closedMessageTask = TestApplicationErrorLogger.WaitForMessage(m => m.Message.Contains("is closed. The last processed stream ID was 1.")).DefaultTimeout();
                 cts.Cancel();
-                try
-                {
-                    await stopServerTask;
-                }
-                // Remove when https://github.com/dotnet/runtime/issues/40290 is fixed
-                catch (OperationCanceledException)
-                {
+                await stopServerTask;
 
-                }
+                // Wait for "is closed" message as this is logged from a different thread and aborting
+                // can timeout and return from server.StopAsync before this is logged.
+                await closedMessageTask;
```

Comment:
Now that we're waiting for the "is closed" log, can we move the `requestUnblocked.SetResult();` so it's called before stopServerTask completes?

```
cts.Cancel();
await closedMessageTask;
requestUnblocked.SetResult();
await stopServerTask;
```

This test doesn't need to block the request until after the abort times out. Waiting for the abort to start (which is done by `await closedMessageTask`) should be enough.

Reply:
The idea is that once we do that and then increase the abort timeout, we won't need to have `new DiagnosticMemoryPoolFactory(allowLateReturn: true);` in this test. Controlling the abort timeout from the test is kind of a big change, but unblocking the request before StopAsync completes is easy enough.